make-workspace-artifact/action.yml

```diff
@@ -1,18 +1,30 @@
 name: "Make workspace artifact"
+description: "This action creates an artifact for the selected workspace."
 
 inputs:
   workspace-name:
+    description: The name of the workspace to create the artifact for"
     required: true
+  workspace-type:
+    description: |
+      The type of the workspace to create the artifact for.
+      Will be inferred if not provided.
+      The supported workspace types are: function-app, next-standalone, package
+    required: false
 outputs:
   artifact-path:
+    description: The path to the created artifact
     value: ${{ steps[format('make-{0}-artifact', steps.detect-workspace-type.outputs.workspace-type)].outputs.artifact-path }}
 runs:
   using: "composite"
   steps:
-    - name: Detect workspace type
-      id: detect-workspace-type
+    # If not provided, try to detect the type of the workspace from its content.
+    - name: Set workspace type
```

Comment:
What do you think if we split these steps? We can use the `if` property of the step to execute `set-workspace-type` or `detect-workspace-type` depending on `inputs.workspace-type`

Reply:
It was my first implementation:
```yaml
     # Set the workspace type if provided in input (it's just a value passthrough)
    - name: Set workspace type
      id: set-workspace-type
      if: ${{ !inputs.workspace-type }}
      shell: bash
      run: |
        echo "workspace-type=${{ inputs.workspace-type }}" >> "$GITHUB_OUTPUT"
    # If not provided, try to detect the type of the workspace from its content.
    - name: Detect workspace type
      id: detect-workspace-type
      if: ${{ !inputs.workspace-type }}
      run: |
        if [ -e "host.json" ]; then
          echo "::debug:: the workspace contains an azure function app"
          echo "workspace-type=function-app" >> "$GITHUB_OUTPUT"
        elif [ -e "next.config.js" ] && grep -q 'output: "standalone"' "next.config.js"; then
          echo "::debug:: the workspace contains a next standalone app"
          echo "workspace-type=next-standalone" >> "$GITHUB_OUTPUT"
        else
          echo "::error:: workspace not supported (should be a azure function app or a next standalone application)"
          exit 1
        fi
      shell: bash
      working-directory: ./out/apps/${{ inputs.workspace-name }}
```

In the end, I thought it was _better_ with just one step. But I haven't a strong opinion on that, if you think is better to split then I just revert the refactoring. Let me know.

Reply:
I prefer the splitted version, I think is more maintainable

Reply:
@lucacavallaro I changed my mind. Values will be then referenced as output from the step. If we assign the `workflow-type` variable in more than one step, how could we reference it? The option would be to `export` variables instead of appending them to the GITHUB_OUTPUT.

I'd keep the cose as is